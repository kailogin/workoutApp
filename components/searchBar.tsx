import { SetStateAction, Dispatch } from "react";
import { StyleSheet, TextInput, View, Keyboard } from "react-native";
import { Feather, Entypo } from "@expo/vector-icons";
import { Colors } from "../utils/colors";

interface SearchBarProps {
  isSearchBarClicked: boolean;
  searchPhrase: string;
  setSearchPhrase: Dispatch<SetStateAction<string>>;
  setIsSearchBarClicked: Dispatch<SetStateAction<boolean>>;
}

export const SearchBar = ({
  isSearchBarClicked,
  searchPhrase,
  setSearchPhrase,
  setIsSearchBarClicked,
}: SearchBarProps) => {
  // --- RENDER ---

  return (
    <View style={styles.container}>
      <View style={styles.searchBar}>
        <Feather
          color="black"
          name="search"
          size={20}
          style={{ marginLeft: 1 }}
        />

        <TextInput
          onChangeText={setSearchPhrase}
          onFocus={() => {
            setIsSearchBarClicked(true);
          }}
          placeholder="Search"
          style={isSearchBarClicked ? styles.inputClicked : styles.input}
          value={searchPhrase}
        />

        {isSearchBarClicked && (
          <Entypo
            color="black"
            name="cross"
            onPress={() => {
              setSearchPhrase("");
              Keyboard.dismiss();
              setIsSearchBarClicked(false);
            }}
            size={20}
            style={{ padding: 1 }}
          />
        )}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    flexDirection: "row",
    justifyContent: "flex-start",
    margin: 40,
    width: "90%",
  },
  searchBar: {
    alignItems: "center",
    backgroundColor: Colors.WHITE,
    borderRadius: 15,
    flexDirection: "row",
    padding: 10,
    width: "95%",
  },
  inputClicked: {
    borderColor: "#F2994A",
    fontSize: 20,
    marginLeft: 10,
    width: "90%",
  },
  input: {
    fontSize: 20,
    marginLeft: 10,
    width: "90%",
  },
});
