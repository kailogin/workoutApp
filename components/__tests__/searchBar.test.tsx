import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { SearchBar } from "../searchBar";

test(`SearchBar snapshot`, () => {
  const snap = renderer
    .create(
      <SearchBar
        isSearchBarClicked={true}
        searchPhrase="TEST"
        setSearchPhrase={() => {}}
        setIsSearchBarClicked={() => {}}
      />
    )
    .toJSON();

  expect(snap).toMatchSnapshot();
});
