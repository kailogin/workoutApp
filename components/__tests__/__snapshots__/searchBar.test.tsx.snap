// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "marginBottom": 40,
      "width": "90%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderRadius": 15,
        "flexDirection": "row",
        "padding": 10,
        "width": "95%",
      }
    }
  >
    <Text />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="searchBarPlaceholder"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#F2994A",
          "fontSize": 16,
          "marginLeft": 10,
          "width": "90%",
        }
      }
      underlineColorAndroid="transparent"
      value="TEST"
    />
    <Text />
  </View>
</View>
`;
